src/utils/get-base-and-head-commit-shas.ts

```diff
@@ -30,3 +37,55 @@ export const getBaseAndHeadCommitShas = async (
 const assertNever = (event: never): never => {
   throw new Error("Unexpected event: " + JSON.stringify(event));
 };
+
+const tryGetMergeCommitBase = (headSha: string): string | null => {
+  const mergeCommitSha = process.env.GITHUB_SHA;
```

Comment:
At this point we assume that the current checked-out version will be the merge commit (`process.env.GITHUB_SHA`) which I think is true in most cases. 

In cases where this isn't true (for example if checking out the top of head) this will error and will fallback to base of PR which I think is sensible. I wonder if it's worth documenting this explicitly here?

An alternative might be to use `git rev-list --max-count=1 HEAD` instead of `process.env.GITHUB_SHA`, which might add more clarity here.

Reply:
Yeah that's a good call. I'll make this explicit.